rosette/src/Import.cc

```diff
@@ -19,12 +20,188 @@
+        break;
+    case ObjectCodePB::OT_STD_METHOD:
+        retval = StdMthd::create(NULL, NULL, NULL);
+        break;
+    case ObjectCodePB::OT_SYMBOL:
+        retval = SYMBOL(ob.symbol().name().c_str());
+        break;
+    case ObjectCodePB::OT_TBL_OBJECT:
+        retval = TblObject::create();
+        break;
+    case ObjectCodePB::OT_TEMPLATE:
+        retval = Template::create(NIL, NILmeta, NILpattern);
+        break;
+    case ObjectCodePB::OT_TUPLE:
+        retval = Tuple::create();
+        break;
+    case ObjectCodePB::OT_TUPLE_EXPR:
+        retval = TupleExpr::create();
+        break;
+    case ObjectCodePB::OT_RBL_BOOL:
+        retval = RBLBOOL(ob.rbl_bool().value());
+        break;
+    case ObjectCodePB::OT_NIV:
+        retval = NIV;
+        break;
+    case ObjectCodePB::OT_STD_EXTENSION:
+        retval = StdExtension::create( ob.std_extension().elements_size());
+        break;
+    case ObjectCodePB::OT_RBL_ATOM:
+        // ???
+        break;
+    case ObjectCodePB::OT_NULL_EXPR:
+        retval = NullExpr::create();
+        break;
+    case ObjectCodePB::OT_COMPOUND_PATTERN:
+        // ???
+        break;
+    case ObjectCodePB::OT_COMPLEX_PATTERN:
+        retval = ComplexPattern::create(NULL);
+        break;
+    case ObjectCodePB::OT_STD_META:
+        retval = StdMeta::create(NIL);
+        break;
+    case ObjectCodePB::OT_ID_VEC_PATTERN:
+        retval = IdVecPattern::create(NILexpr);
+        break;
+    case ObjectCodePB::OT_ID_AMPR_REST_PATTERN:
+        retval = IdAmperRestPattern::create(NULL);
+        break;
+    case ObjectCodePB::OT_ID_PATTERN:
+        retval = IdPattern::create( SYMBOL(ob.id_pattern().symbol().symbol().name().c_str()) );
+        break;
+    case ObjectCodePB::OT_PRIM:
+        retval =  Prim::create(
+            (char *)ob.prim().id().symbol().name().c_str(),
+            NULL,
+            (int)ob.prim().minargs(),
+            (int)ob.prim().maxargs());
+        break;
+    case ObjectCodePB::OT_SYSVAL: {
+        std::string val = ob.sysval().value();
+
+        if (val == "#inv")
+            retval = INVALID;
+        else if (val == "#upcall")
+            retval = UPCALL;
+        else if (val == "#suspended")
+            retval = SUSPENDED;
+        else if (val == "#interrupt")
+            retval = INTERRUPT;
+        else if (val == "#sleep")
+            retval = SLEEP;
+        else if (val == "#deadThread")
+            retval = DEADTHREAD;
+        break;
+    }
+    case ObjectCodePB::OT_CONST_PATTERN:
+        retval = ConstPattern::create(FIXNUM(ob.const_pattern().val().fixnum().value()));
+        break;
+    case ObjectCodePB::OT_RBL_TABLE:
+        retval = RblTable::create();
+        break;
+    default:
+        warning("Import object type=%d not yet implemented!", ob.type());
+    }
+
+    return retval;
+}
+
+pOb linkRosetteObject(ObjectInfo & obi) {
```

Comment:
What does this do?

Reply:
This isn't written yet, but it will go through all the objects, look up referenced objects by ID, and then put their pointers into the appropriate places.